File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        
        for(int i=0; i<clothes.length; i++) {
            String category = clothes[i][1];
            
            if(map.containsKey(category)) {
                map.put(category, map.get(category) + 1);
                continue;
            }
            
            map.put(category, 1);
        }
                
        int answer = 1;
        for(var entry : map.entrySet()) {
            answer *= (entry.getValue() + 1);
        }
    
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string,int> m;
    for(vector<string> cloth : clothes) {
        string name = cloth[0];
        string category = cloth[1];
        
        m[category]++;
    }
    
    int tmp = 1;
    for(auto [k, v] : m) {
        tmp *= (v + 1);
        
    }
    return tmp - 1;
}
